### PR TITLE
Add Script to Sample Sentences for Tokenizer Training

### PR DESCRIPTION
**Pull request by ToluClassics**
Sample sentences from monolingual corpus based on implentation here 

section 3.1 here [https://arxiv.org/pdf/1901.07291.pdf](https://arxiv.org/pdf/1901.07291.pdf)

### CODE REVIEW

File: scripts/sample_tokenizer_sentences.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        lang_code = file.split(".")[-1]
+        with open(file) as f:
+            txt = f.readlines()
+            lang_corpus[lang_code] = txt
+            lang_num_lines[lang_code] = len(txt)
+
+    sampled_sentences = calc_num_samples_sentences(lang_num_lines, args.alpha)
+
+    for lang in tqdm(sampled_sentences.keys()):
+        lines = random.sample(range(lang_num_lines[lang]), sampled_sentences[lang])
```
Comment (keleog) [inline scripts/sample_tokenizer_sentences.py]: we should seed the random call before sampling so we can always reproduce.

File: scripts/sample_tokenizer_sentences.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        file = os.path.join(args.output_path, "sampled." + lang)
+
+        with open(file, "w") as out_file:
+            sentences = list(map(lambda i: lang_corpus[lang][i], lines))
```
Comment (keleog) [inline scripts/sample_tokenizer_sentences.py]: i am a bit confused about what this does and why it is necessary.

File: scripts/sample_tokenizer_sentences.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import os
+import argparse
+import random
+import subprocess
```
Comment (keleog) [inline scripts/sample_tokenizer_sentences.py]: unused import

File: scripts/sample_tokenizer_sentences.py (hunk 2 above)
Comment (keleog) [reply scripts/sample_tokenizer_sentences.py]: On a second look, I understand now. But why do we decide to sample indices as opposed to sampling sentences directly?

File: scripts/sample_tokenizer_sentences.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    )
+    parser.add_argument(
+        "--output_path", type=str, required=True, help="path to store sampled sentences"
+    )
+    parser.add_argument("--alpha", type=float, default=0.3, help="multinomial alpha")
+
+    return parser
+
+
+def calc_num_samples_sentences(lang_num_lines: dict, alpha: float):
```
Comment (keleog) [inline scripts/sample_tokenizer_sentences.py]: `def calc_num_samples_sentences(lang_num_lines: Dict[str, int], alpha: float) -> Dict[str, int]:`

File: scripts/sample_tokenizer_sentences.py (hunk 2 above)
Comment (ToluClassics) [reply scripts/sample_tokenizer_sentences.py]: fixed!

Comment (keleog) [reply scripts/sample_tokenizer_sentences.py]: nice, thanks!

**Pull request merged**